transport/rif_comms/client.py

```diff
@@ -8,95 +8,101 @@
         """
         self.rsk_address = RskAddress(address=rsk_address)
-        self.grpc_channel = insecure_channel(grpc_api_endpoint)
+        self.grpc_channel = insecure_channel(grpc_api_endpoint)  # TODO: how to make this secure?
         self.stub = CommunicationsApiStub(self.grpc_channel)
 
     def connect(self) -> Notification:
         """
         Connects to RIF Communications Node.
-        Invokes ConnectToCommunicationsNode grpc api endpoint.
-        :return: Notification stream
+        Invokes ConnectToCommunicationsNode GRPC API endpoint.
+        Adds the client RSK address under the RIF Communications node peer ID.
+        :return: notification stream
         """
         return self.stub.ConnectToCommunicationsNode(self.rsk_address)
 
-    def subscribe(self, topic_id: str) -> Notification:
+    def subscribe_to(self, rsk_address: Address) -> Notification:
         """
-        Subscribes to a pub-sub topic between self.rsk_address and partner_address.
-        Invokes CreateTopicWithRskAddress grpc api endpoint.
-        :param topic_id: ID of the topic to subscribe
-        :return: Notification stream
+        Subscribes to a pub-sub topic in order to send messages to or receive messages from an address.
+        Invokes CreateTopicWithRskAddress GRPC API endpoint.
+        The resulting notification stream should only be used for receiving messages; use send_message for sending.
+        :param rsk_address: destination RSK address for message sending
+        :return: notification stream for receiving messages
         """
-        # TODO catch already subscribed and any error
-        return self.stub.CreateTopicWithRskAddress(RskAddress(address=topic_id))
+        # TODO: catch already subscribed and any error
+        return self.stub.CreateTopicWithRskAddress(RskAddress(address=rsk_address))
 
-    # TODO review params and create docstring. It has no sense to use both peer id and rsk_address
-    def has_subscription(self, rsk_address: Address) -> BooleanResponse:
-        peer_id = self.get_peer_id(rsk_address)
+    def is_subscribed_to(self, rsk_address: Address) -> bool:
+        """
+        Returns whether or not the client's underlying RIF Communications node is subscribed to the topic
+        which corresponds to the given RSK address.
+        Invokes HasSubscriber GRPC API endpoint.
+        :param rsk_address: RSK address which corresponds to the topic which is being checked for subscription
+        :return: boolean value indicating whether the client is subscribed or not
+        """
+        our_peer_id = self._get_peer_id(self.rsk_address)
+        topic_id = self._get_peer_id(rsk_address)
         return self.stub.HasSubscriber(
             Subscriber(
-                peerId=peer_id,
-                channel=Channel(channelId=rsk_address)
+                peerId=our_peer_id,
+                channel=Channel(channelId=topic_id)
             )
-        )
+        ).value
```

Comment:
I think that using our_peer_id instead of retrieving the peer id from the rsk_address is ok

You can receive an rsk address that isnt "our" rsk address and try to check with HasSubscriber with a relation of peerid-address that doesnt make sense

also not sure about the channelId=topic_id (maybe it expects the rsk address here)

this must be double checked with rif comms team i guess

Reply:
> You can receive an rsk address that isnt "our" rsk address and try to check with HasSubscriber with a relation of peerid-address that doesnt make sense

would we ever have such a use case? as far as i'm concerned, the node only cares about what the client is subscribed to, not other peers. i might be wrong though.

> also not sure about the channelId=topic_id (maybe it expects the rsk address here)

> this must be double checked with rif comms team i guess

will get back to you on this :+1:

Reply:
the RIF comms team has confirmed that this call will check whether `peer_id` is subscribed to `channel_id`, and that it doesn't specifically work with RSK addresses.